spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/SparkSQLExecutionHelper.java

```diff
@@ -42,28 +45,26 @@ public static String lastExecutedMetricValue(SparkSession spark, String metricNa
     SQLExecutionUIData lastExecution = statusStore.executionsList().last();
     Option<SQLPlanMetric> sqlPlanMetric =
         lastExecution.metrics().find(metric -> metric.name().equals(metricName));
-    Assert.assertTrue(
-        String.format("Metric '%s' not found in last execution", metricName),
-        sqlPlanMetric.isDefined());
+    assertThat(sqlPlanMetric.isDefined())
+        .as(String.format("Metric '%s' not found in last execution", metricName))
+        .isTrue();
     long metricId = sqlPlanMetric.get().accumulatorId();
 
     // Refresh metricValues, they will remain null until the execution is complete and metrics are
     // aggregated
-    int attempts = 3;
-    while (lastExecution.metricValues() == null && attempts > 0) {
-      try {
-        Thread.sleep(100);
-        attempts--;
-      } catch (InterruptedException e) {
-        throw new RuntimeException(e);
-      }
+    Awaitility.await()
```

Comment:
nit: is it a good idea to add some alias with `Awaitility` here?

Reply:
in certain cases it makes sense to add an alias, but here I think we should be ok, since there's a comment right above that explains what's being done